HW04.Rmd

```diff
@@ -28,34 +35,112 @@ library(tidyverse)    # load tidyverse packages, including ggplot2
 
 ```
 
 
 ### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
 
-```{r month_chart}
+```{r month_chart, echo=F}
+
+#Since deaths_month$month was already a factor with abbreviated month names (line 47), you only need to graph.
+#The instruction was for the total number of deaths per month, not per year, but the data frame contains observations from 2012 to 2014.
 
+deaths_month %>%
+    ggplot(aes(x = month, y = Deaths_per_month)) + 
+    geom_col(fill = "blue") + #Filled just to pop some color there, I don't like gray.
+        labs(title = "Number of gun deaths per month from 2012-2014", #The instruction was for the total number of deaths per month, not per year, but the data frame contains observations from 2012 to 2014.
+         x = "Month",
+         y = "Number of gun deaths",
+         caption = "Source: fivethirtyeight.com")
 ```
 
 ## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
 
-```{r intent}
+```{r intent, echo=F}
+
+deaths_intent <- guns %>%
+    group_by(intent) %>%
+    count(intent, sort = TRUE) %>% #Sorted values from highest to lowest
+    rename(no_of_deaths = n)
+
+
+kable(deaths_intent, format = "markdown", col.names = c("Intent", "No. of deaths"))
+
+#Just to order the intent from highest to lowest in a factor
+deaths_intent$intent_ordered <- deaths_intent$intent
+deaths_intent$intent <- factor(deaths_intent$intent, levels = deaths_intent$intent_ordered, ordered = TRUE)
+
+
+ggplot(deaths_intent, aes(x = intent, y = no_of_deaths)) +
+    geom_col(fill = "red") + 
+    scale_y_log10() + #Accidental, Undetermined, and NA went almost unnoticed in a regular scale.
+    annotation_logticks(sides = "l") +
+    labs(title = "Number of gun deaths associated with each type of \nintent cause of death from 2012-2014",
+         x = "Intent cause of death",
+         y = "Number of gun deaths (log scale)",
+         caption = "Source: fivethirtyeight.com")
+
 
 ```
 
 ## Generating a boxplot visualizing the age of gun death victims, by sex. Print the average age of female gun death victims.
 
-```{r age_&_sex}
+```{r age_&_sex, echo=F}
+
+#This way we can clean our data from Not Available age values (strings). Also is not strictly necessary, but I wanted to try the select() function.
+death_sex_age <- guns %>%
+    select(sex, age) %>%
+    filter(age != "Not Available")
+
+#We need to change the column type from "character" to "integer" to create the boxplot
+death_sex_age$age <- as.integer(death_sex_age$age)
+
+death_sex_age %>%
+    ggplot(aes(x = sex, y = age, fill = sex), alpha = 0.5) +
+    geom_boxplot() +
+    labs(title = "Age distibution of gun death victims by sex",
+         x = "Sex",
+         y = "Age",
+         caption = "Source: fivethirtyeight.com")

```

Comment:
Great plot, I am surprised by how similar the distributions are for males and females (except the outliers for females) - I think the aesthetics of the outliers is a bit muddled. Would there be a way to pick a new shape/color + outline for them so they stand out as individual points more?

Reply:
Well, I don't know how to, but I think it's worth trying, it's a good idea.